src/components/spreadsheet/table-wrapper.js

```diff
@@ -0,0 +1,686 @@
+            enrichColumn,
+            isEditColumnVisible,
+            reorderedTableDefinitionIndexes,
+            selectedColumnsNames,
+        ]
+    );
+
+    const getRows = useCallback(
+        (index) => {
+            if (props.disabled || !props.network) {
+                return [];
+            }
+            const tableDefinition = TABLES_DEFINITION_INDEXES.get(index);
+            const datasourceRows = tableDefinition.getter
+                ? tableDefinition.getter(props.network)
+                : props.network[TABLES_DEFINITION_INDEXES.get(index).resource];
+
+            if (!datasourceRows) return [];
+            return datasourceRows;
+        },
+        [props.disabled, props.network]
+    );
+
+    useEffect(() => {
+        setColumnData(generateTableColumns(tabIndex));
+        setRowData(getRows(tabIndex));
+    }, [generateTableColumns, getRows, tabIndex]);
+
+    const handleSwitchTab = useCallback(
+        (value) => {
+            setManualTabSwitch(true);
+            setTabIndex(value);
+            cleanTableState();
+        },
+        [cleanTableState]
+    );
+
+    useEffect(() => {
+        const allDisplayedTemp = allDisplayedColumnsNames[tabIndex];
+        const newSelectedColumns = new Set(
+            allDisplayedTemp ? JSON.parse(allDisplayedTemp) : []
+        );
+        setSelectedColumnsNames(newSelectedColumns);
+    }, [tabIndex, allDisplayedColumnsNames]);
+
+    useEffect(() => {
+        const allLockedTemp = allLockedColumnsNames[tabIndex];
+        setLockedColumnsNames(
+            new Set(allLockedTemp ? JSON.parse(allLockedTemp) : [])
+        );
+    }, [tabIndex, allLockedColumnsNames]);
+
+    useEffect(() => {
+        const allReorderedTemp = allReorderedTableDefinitionIndexes[tabIndex];
+        setReorderedTableDefinitionIndexes(
+            allReorderedTemp
+                ? JSON.parse(allReorderedTemp)
+                : TABLES_DEFINITION_INDEXES.get(tabIndex).columns.map(
+                      (item) => item.id
+                  )
+        );
+    }, [allReorderedTableDefinitionIndexes, tabIndex]);
+
+    useEffect(() => {
+        const resource = TABLES_DEFINITION_INDEXES.get(tabIndex).resource;
+        if (!props.network || props.disabled) return;
+        props.network.useEquipment(resource);
+    }, [props.network, props.disabled, tabIndex]);
+
+    useEffect(() => {
+        setManualTabSwitch(false);
+    }, [props.equipmentChanged]);
+
+    function getTabIndexFromEquipementType(equipmentType) {
+        const definition = Object.values(TABLES_DEFINITIONS).find(
+            (d) => d.name.toLowerCase() === equipmentType.toLowerCase()
+        );
+        return definition ? definition.index : 0;
+    }
+
+    useEffect(() => {
+        if (
+            props.equipmentId !== null &&
+            props.equipmentType !== null &&
+            !manualTabSwitch
+        ) {
+            const newIndex = getTabIndexFromEquipementType(props.equipmentType);
```

Comment:
May be name it (new/next)TabIndex and the row index ... (new/next)RowIndex

Reply:
Done, i've also removed the call to getRows and directly use rowData instead which already contains the data needed